contracts/UFragments.sol

```diff
@@ -255,4 +291,53 @@ contract UFragments is ERC20Detailed, Ownable {
         emit Approval(msg.sender, spender, _allowedFragments[msg.sender][spender]);
         return true;
     }
+
+    /**
+     * @dev Allows for approvals to be made via secp256k1 signatures.
+     * @param owner The owner of the funds
+     * @param spender The spender
+     * @param value The amount
+     * @param deadline The deadline timestamp, type(uint256).max for max deadline
+     * @param v Signature param
+     * @param s Signature param
+     * @param r Signature param
+     */
+    function permit(
+        address owner,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) public {
+        //solium-disable-next-line
+        require(block.timestamp <= deadline);
+
+        uint256 currentValidNonce = _nonces[owner];
+        bytes32 digest =
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    DOMAIN_SEPARATOR,
+                    keccak256(
+                        abi.encode(
+                            PERMIT_TYPEHASH,
+                            owner,
+                            spender,
+                            value,
+                            currentValidNonce,
+                            deadline
+                        )
+                    )
+                )
+            );
+
+        require(owner == ecrecover(digest, v, r, s));
```

Comment:
add revert message?

Reply:
Dont think revert messages are possible in 0.4.24